tests/integration_test.rs

```diff
@@ -451,6 +451,105 @@ fn test_more_complex_program_with_conditionals() {
+    let (result, _) = run_program_with_custom_state(&bin_path, vm_with_custom_flags);
+
+    assert_eq!(result, U256::from(0b1001));
+}
+
+#[test]
+fn test_xor_asm() {
+    let bin_path = make_bin_path_asm("xor");
+    let r1 = U256::from(0b1011);
+    let r2 = U256::from(0b1101);
+    let mut registers: [U256; 15] = [U256::zero(); 15];
+    registers[0] = r1;
+    registers[1] = r2;
+    let vm_with_custom_flags = VMStateBuilder::new().with_registers(registers).build();
+    let (result, _) = run_program_with_custom_state(&bin_path, vm_with_custom_flags);
+
+    assert_eq!(result, U256::from(0b0110));
+}
+
+#[test]
+fn test_or_asm() {
+    let bin_path = make_bin_path_asm("or");
+    let r1 = U256::from(0b1011);
+    let r2 = U256::from(0b1101);
+    let mut registers: [U256; 15] = [U256::zero(); 15];
+    registers[0] = r1;
+    registers[1] = r2;
+    let vm_with_custom_flags = VMStateBuilder::new().with_registers(registers).build();
+    let (result, _) = run_program_with_custom_state(&bin_path, vm_with_custom_flags);
+
+    assert_eq!(result, U256::from(0b1111));
+}
+
+#[test]
+fn test_jump_asm() {
+    let bin_path = make_bin_path_asm("jump");
+    let (result, _) = run_program(&bin_path);
+    assert_eq!(result, U256::from(42));
+}
+
+#[test]
+fn test_jump_label() {
+    let bin_path = make_bin_path_asm("jump_label");
+    let (result, _) = run_program(&bin_path);
+    assert_eq!(result, U256::from(42));
+}
+
+#[test]
+fn test_and_conditional_jump() {
+    let bin_path = make_bin_path_asm("and_conditional_jump");
+    let r1 = U256::from(0b1011);
+    let r2 = U256::from(0b1101);
+    let mut registers: [U256; 15] = [U256::zero(); 15];
+    registers[0] = r1;
+    registers[1] = r2;
+    let vm_with_custom_flags = VMStateBuilder::new().with_registers(registers).build();
+    let (result, _) = run_program_with_custom_state(&bin_path, vm_with_custom_flags);
+
+    assert_eq!(result, U256::from(0b1001));
+}
+
+#[test]
+fn test_xor_conditional_jump() {
+    let bin_path = make_bin_path_asm("xor_conditional_jump");
+    let r1 = U256::from(0b1011);
+    let r2 = U256::from(0b1101);
+    let mut registers: [U256; 15] = [U256::zero(); 15];
+    registers[0] = r1;
+    registers[1] = r2;
+    let vm_with_custom_flags = VMStateBuilder::new().with_registers(registers).build();
+    let (result, _) = run_program_with_custom_state(&bin_path, vm_with_custom_flags);
+
+    assert_eq!(result, U256::from(0b0110));
+}
+
+#[test]
+fn test_or_conditional_jump() {
+    let bin_path = make_bin_path_asm("or_conditional_jump");
+    let r1 = U256::from(0b1011);
+    let r2 = U256::from(0b1101);
+    let mut registers: [U256; 15] = [U256::zero(); 15];
+    registers[0] = r1;
+    registers[1] = r2;
+    let vm_with_custom_flags = VMStateBuilder::new().with_registers(registers).build();
+    let (result, _) = run_program_with_custom_state(&bin_path, vm_with_custom_flags);
+
+    assert_eq!(result, U256::from(0b1111));
```

Comment:
Jump instructions set a new PC, right?
Shouldn't we also check that on the tests?

Reply:
You are right, modified the jump label test so it checks this:
```rs
#[test]
fn test_jump_label() {
    let bin_path = make_bin_path_asm("jump_label");
    let (result, vm_final_state) = run_program(&bin_path);
    let final_pc = vm_final_state.current_frame.pc;
    assert_eq!(result, U256::from(42));
    // failing to jump into the label will finish program with pc == 2
    assert_eq!(final_pc, 6)
}
```